src/metadata/metadata.c

```diff
@@ -27,7 +27,10 @@
 #include "dislocker/metadata/metadata_config.h"
 #include "dislocker/metadata/print_metadata.h"
 #include "dislocker/dislocker.priv.h"
-#include <sys/mount.h>
+
+#include <sys/ioctl.h>
+
+#define BLKSSZGET  _IO(0x12,104)/* get block device sector size */
```

Comment:
I just realized that calling [ioctl()](https://github.com/Aorimn/dislocker/blob/6a653083adf17f9511fc631ab28e9770c80056d3/src/metadata/metadata.c#L170) may be proone to errors: will it return the expected result when the `fve_fd` points to a file (e.g. when a partition has been dumped), not to a volume/partition?

Reply:
When “ioctl()” fails to get sector size, by default 512 bytes will be set as the default sector size. Because dbr does not record sector size if the exFAT BitLocker drive is created on Windows10 1903

https://github.com/Aorimn/dislocker/blob/6a653083adf17f9511fc631ab28e9770c80056d3/src/metadata/metadata.c   167-175 line

//Windows 10 1903 exFAT
if (!dis_meta->volume_header->sector_size) {
	uint64_t nSectorSize = 0;
	ioctl(dis_meta_cfg->fve_fd, BLKSSZGET, &nSectorSize);
	if(!nSectorSize)
		nSectorSize = 512;
	dis_meta->volume_header->sector_size = (uint16_t)nSectorSize;
}
//Windows 10 1903 exFAT